### ISSUE TITLE
Detect inactive people in Bugzilla

### ISSUE THREAD

**Issue by suhaibmujahid** [mozilla/relman-auto-nag#1321]
Multiple issues require detecting inactive users to perform an action or produce a report, e.g.:
- #190
- #266
- #1231
- #1179

The question is, how we should proceed in detecting inactive users?
 
@calixteman suggested using the diff in the `people.json` file to know who left (works for Mozilla employees only).

In [assignee_no_login.py#L72](https://github.com/marco-c/relman-auto-nag/blob/7c9e5c953d62521961e76155d5a55318c3f0c22c/auto_nag/scripts/assignee_no_login.py#L72), the `assignee_last_login` column is used to determine inactive assignees. However, based on a comment by Kohei, it seems that the `assignee_last_login` column does not give the desired signal.

---

> https://bugzilla.mozilla.org/show_bug.cgi?id=1527841#c3:
> ### Kohei Yoshino - comment 3
> I’ve realized that the `assignee_last_login` column is useless. It’s literally the last login date of the user, not the last seen date in spite of the internal field name called `last_seen_date` , which is perhaps a legacy of the days when people had logged into Bugzilla everyday at 9am and shut down their workstation at 5pm. Now people keep logged in, so the date can be quite old.
> 
> We need the new `assignee_last_seen` column based on another internal field called `last_activity_ts`, which is currently a part of the UserProfile extension. I’ll file a separate bug for that.
> 
> The same applies to needinfo. What we need here is not `needinfo_last_login` but `needinfo_last_seen`, right?

**Comment by marco-c** [comment]
Bugzilla still periodically requests to re-login, so if we set a high enough threshold we should be fine (indeed, so far the assignee_no_login didn't find any inactive user that was actually active).

We can then switch to a better method later.

**Comment by marco-c** [comment]
A third way we can use is the Bugzilla User REST API (https://bugzilla.readthedocs.io/en/latest/api/core/v1/user.html), checking if an account is disabled.
A disabled account can be considered an inactive account.

**Comment by marco-c** [comment]
#1342 is implementing most of this, except for the identification of people who left Mozilla through the usage of people.json (see also #1354 which is related).

**Comment by marco-c** [comment]
> #1342 is implementing most of this, except for the identification of people who left Mozilla through the usage of people.json (see also #1354 which is related).

@suhaibmujahid there is still this left to do after #1342 is fixed, do you want to file a new issue or reopen this one?

**Comment by suhaibmujahid** [comment]
@marco-c the  `people.json` part is already covered with #1354. But I will fill a new issue to breakdown #1354.

**Comment by suhaibmujahid** [comment]
I filed #1371 to tack the `people.json` part.
